firrtl/src/main/scala/firrtl/ir/IR.scala

```diff
@@ -543,7 +545,6 @@ case class IntModule(
     extends DefModule
     with UseSerializer
 
-case class Circuit(info: Info, modules: Seq[DefModule], main: String, annotations: AnnotationSeq)
-    extends FirrtlNode
-    with HasInfo
-    with UseSerializer
+case class Circuit(info: Info, modules: Seq[DefModule], main: String) extends FirrtlNode with HasInfo with UseSerializer
+
+case class CircuitWithAnnos(circuit: Circuit, annotations: Seq[Annotation]) extends FirrtlNode with UseSerializer
```

Comment:
Any reason to break this up as opposed to keeping it unified?

Reply:
Yeah because the Converter creates a firrtl Circuit but returns Seq[Annotation]. I'm totally down with refactoring/replacing/nuking the Stage/Phase approach but this is a bit more of a min-cut approach that keeps all Stages and Phases working as they have in the past (including custom user code filtering out annotations after running the Converter which we're doing at SiFive) while only imposing the new "CircuitWithAnnos" upon emission at the end of the Stage. In any case, let's rewrite it all but I think for 5.0 this is an appropriately modest way to do it.